### PR TITLE
band edit and destroy features

### PR DESCRIPTION
**Pull request by RayReeves**

### CODE REVIEW

File: app/controllers/api/v1/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Api::V1::ReviewsController < ApplicationController
+  
+  def create
+    review = Review.new(review_params)
+    binding.pry
+    
+  end
```
Comment (dovakeen118) [inline app/controllers/api/v1/reviews_controller.rb]: PRY! and empty line in the create action

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Api::V1::UsersController < ApplicationController
+
+ def index
+  binding.pry
+  render json: User.find(current_user.id)
```
Comment (dovakeen118) [inline app/controllers/api/v1/users_controller.rb]: PRY!

File: app/controllers/bands_controller.rb [hunk 3]
```diff
@@ -15,6 +15,33 @@ def create
...
+  def edit
+    @band = Band.find(params[:id])
+  end
+
+  def update
+    @band = Band.find(params[:id])
+
+    if @band.update(band_params)
+      flash[:notification] = "Changes have been made"
+      redirect_to root_path
```
Comment (dovakeen118) [inline app/controllers/bands_controller.rb]: Would we possibly want to be redirected back to the show page for the band that was just updated?

File: app/controllers/bands_controller.rb [hunk 4]
```diff
@@ -15,6 +15,33 @@ def create
...
+    else
+      flash.now[:error] = @band.errors.full_messages.to_sentence
+      render :edit
+    end
+  end
+
+  def destroy
+    @band = Band.find(params[:id])
```
Comment (dovakeen118) [inline app/controllers/bands_controller.rb:38]: With more actions now would be a good time to discuss with your group about restricting user access, as in only allowing signed in users to perform actions involving creation/modification/destruction. This also ensures that only the user that created a resource as the ability to update/delete it!

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Nice work so far, some clean up do to!

File: app/controllers/bands_controller.rb (hunk 3 above)
Comment (nwalberts) [reply app/controllers/bands_controller.rb]: I would encourage you to tackle Kerrins comment!

File: app/javascript/react/components/bands/BandShow.js [hunk 5]
```diff
@@ -9,6 +15,7 @@ const BandShow = (props) => {
       <div className="card-divider">
         <p className="card-section">{props.biography}</p>
       </div>
+      {adminLogin()}
```
Comment (nwalberts) [inline app/javascript/react/components/bands/BandShow.js]: close but we dont need a function for this stuff. instead 

```
   let editElement
    if (props.user.role === "admin") {
      editElement = <span><a href={`/bands/${props.id}/edit`}>Edit</a></span>
    } 

    .... 
     {editElement}

```

File: app/javascript/react/components/bands/BandShowContainer.js [hunk 6]
```diff
@@ -25,11 +27,31 @@ const BandShowContainer = (props) => {
...
+  const fetchUser = async () => {
+    try{
+      const response = await fetch(`/api/v1/users`)
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const parsedUserObject = await response.json()
+    setUser(parsedUserObject)
```
Comment (nwalberts) [inline app/javascript/react/components/bands/BandShowContainer.js]: indentation is off here

File: app/views/bands/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  <div>
+    <h3>(Optional): Upload Band Profile Image</h3>
+    <%= f.file_field :image%>
+  </div>
+
+  <%= f.submit "Make Changes"%>
+<%end%>
+</br>
+</br>
```
Comment (nwalberts) [inline app/views/bands/edit.html.erb]: avoid using break tags. instead use CSS!

File: app/views/bands/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+<%=button_to "Delete Band", @band, :method => :delete, data: { confirm: "You sure?" } %>
```
Comment (nwalberts) [inline app/views/bands/edit.html.erb]: im not a huge fan of the prompt, but ill allow it. however `:method => :delete` uses old syntax. Change to 
`method: :delete`

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: app/controllers/bands_controller.rb [hunk 9]
```diff
@@ -15,11 +18,43 @@ def create
...
+  def edit
+    @band = Band.find(params[:id])
+  end
+
+  def update
+    @band = Band.find(params[:id])
+
+    if @band.update(band_params)
+      flash[:notification] = "Changes have been made"
+      redirect_to "/bands/show/#{params[:id]}"
```
Comment (nwalberts) [inline app/controllers/bands_controller.rb]: unfortunately, this path looks wrong, which suggests that maybe you didnt test it...?

`/bands/#{params[:id]}` is more what you want

I would also change the message to "Band updated" or "your changes have been made"

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**